salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +50,94 @@ def node_info(node, ca_minion):
     return result
 
 
+def cluster_version_info(namespace_information):
+    cluster_version = None
+
+    if namespace_information:
+        ns_info_dict = namespace_information.to_dict()
+        annotation_info = ns_info_dict.get(
+            'metadata', {}).get('annotations')
+        if annotation_info:
+            cluster_version = annotation_info.get(
+                'metalk8s.scality.com/cluster-version'
+            )
+    if not cluster_version:
+        cluster_version = __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return cluster_version
+
+
+def get_storageclass(storageClassV1):
```

Comment:
The API to use is an implementation detail of this method IMO, shouldn't be the responsibility of the caller to know which one to pass. I'd suggest a simpler signature:

```python
def list_storage_classes(api_client):
    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
    storage_classes = {}
    ...
```